tty/imx6ull-uart/imx6ull-uart.c

```diff
@@ -29,12 +32,71 @@
 #include <sys/stat.h>
 #include <sys/debug.h>
 #include <posix/utils.h>
+#include <fcntl.h>
+#include <syslog.h>
 
 #include <libtty.h>
 #include <libklog.h>
 
 #include <phoenix/arch/imx6ull.h>
 
+#define COL_RED    "\033[1;31m"
+#define COL_YELLOW "\033[1;33m"
+#define COL_NORMAL "\033[0m"
+
+#define LOG_TAG "imx6ull-uart"
+
+/* clang-format off */
+#define log_debug(fmt, ...) do { log_printf(LOG_DEBUG, fmt "\n", ##__VA_ARGS__); } while (0)
+#define log_info(fmt, ...)  do { log_printf(LOG_INFO, fmt "\n", ##__VA_ARGS__); } while (0)
+#define log_warn(fmt, ...)  do { log_printf(LOG_WARNING, COL_YELLOW fmt COL_NORMAL "\n", ##__VA_ARGS__); } while (0)
+#define log_error(fmt, ...) do { log_printf(LOG_ERR, COL_RED fmt COL_NORMAL "\n", ##__VA_ARGS__); } while (0)
+/* clang-format on */
+
+#define READER 1
+#define WRITER 2
+
+#define MAX_READERS 255
```

Comment:
I thought we want only one reader to be allowed

Reply:
I thought you have requested it. One reader is fine by me. Linux allows multiple readers.

Reply:
Ok, so misunderstanding, I thought we wanted to restrict to one reader only. I see no problem with allowing multiple readers on a driver side, it shouldn't really bother the driver who and how consumes the data.